core/src/main/java/tachyon/client/TachyonFS.java

```diff
@@ -1020,6 +1025,19 @@ public synchronized void updateRawTableMetadata(int id, ByteBuffer metadata) thr
    * @param uri The uri to validate
    */
   private void validateUri(TachyonURI uri) throws IOException {
+    if (!uri.isPathAbsolute() && !TachyonURI.EMPTY_URI.equals(uri)) {
+      LOG.debug("URL is invalid because of Absolute");
+    }
+    if (uri.hasScheme() && !mRootUri.getScheme().equals(uri.getScheme())) {
+      LOG.debug("URL is invalid because of schema");
+    }
+
+    if (uri.hasAuthority() && !mRootUri.getAuthority().equals(uri.getAuthority())) {
+      LOG.debug("URL is invalid because of authority. URI has authoring "
+              + uri.hasAuthority() + " root authoring " + mRootUri.getAuthority()
+              + " uri authority " + uri.getAuthority());
+    }
+
```

Comment:
I meant since the complex sanity checking statement is being separated into different conditions, it would be better to write it as
```
if (uri == null) {
  LOG.debug("URI cannot be null.");
  throw new IOException("Uri " + uri + " is invalid.");
}
if (!uri.isPathAbsolute() && !TachyonURI.EMPTY_URI.equals(uri)) {
  LOG.debug("Uri must have an absolute path.");
  throw new IOException("Uri " + uri + " is invalid.");
}
```
and so on for each of the or-ed conditions that have been separated. Then the existing condition check can be removed.
Also I think there is a typo in some of the debug statements with `URL` instead of `URI`